src/main/scala/se/nimsa/sbx/dicom/streams/AnonymizationFlow.scala

```diff
@@ -201,70 +203,77 @@ object AnonymizationFlow {
+  )
 
   /**
     * From standard PS3.15 Table E.1-1
     * Remove all private attributes
     * Remove overlay data
     * Remove, set empty or modify certain attributes
     */
-  val anonFlow: Flow[DicomPart, DicomPart, NotUsed] = Flow[DicomPart]
-    .via(blacklistFilter(DicomParsing.isPrivateAttribute _)) // remove private attributes
-    .via(blacklistFilter(isOverlay _)) // remove overlay data
-    .via(blacklistFilter(removeTags.contains _)) // remove tags from above list, if present
-    .via(modifyFlow( // modify, clear and insert
-    modify(Tag.AccessionNumber, bytes => if (bytes.nonEmpty) createAccessionNumber(bytes) else bytes),
-    modify(Tag.ConcatenationUID, createUid),
-    clear(Tag.ContentCreatorName),
-    modify(Tag.ContextGroupExtensionCreatorUID, createUid),
-    clear(Tag.ContrastBolusAgent),
-    modify(Tag.CreatorVersionUID, createUid),
-    insert(Tag.DeidentificationMethod, _ => toAsciiBytes("Retain Longitudinal Full Dates Option", VR.LO)),
-    modify(Tag.DimensionOrganizationUID, createUid),
-    modify(Tag.DoseReferenceUID, createUid),
-    modify(Tag.FiducialUID, createUid),
-    clear(Tag.FillerOrderNumberImagingServiceRequest),
-    modify(Tag.FrameOfReferenceUID, _ => createUid(null)),
-    modify(Tag.InstanceCreatorUID, createUid),
-    modify(Tag.IrradiationEventUID, createUid),
-    modify(Tag.LargePaletteColorLookupTableUID, createUid),
-    modify(Tag.MediaStorageSOPInstanceUID, createUid),
-    modify(Tag.ObservationSubjectUIDTrial, createUid),
-    modify(Tag.ObservationUID, createUid),
-    modify(Tag.PaletteColorLookupTableUID, createUid),
-    insert(Tag.PatientIdentityRemoved, _ => toAsciiBytes("YES", VR.CS)),
-    insert(Tag.PatientID, _ => createUid(null)),
-    insert(Tag.PatientName, _ => createUid(null)),
-    clear(Tag.PlacerOrderNumberImagingServiceRequest),
-    modify(Tag.ReferencedFrameOfReferenceUID, createUid),
-    modify(Tag.ReferencedGeneralPurposeScheduledProcedureStepTransactionUID, createUid),
-    modify(Tag.ReferencedObservationUIDTrial, createUid),
-    modify(Tag.ReferencedSOPInstanceUID, createUid),
-    modify(Tag.ReferencedSOPInstanceUIDInFile, createUid),
-    clear(Tag.ReferringPhysicianName),
-    modify(Tag.RelatedFrameOfReferenceUID, createUid),
-    modify(Tag.RequestedSOPInstanceUID, createUid),
-    insert(Tag.SeriesInstanceUID, _ => createUid(null)),
-    insert(Tag.SOPInstanceUID, createUid),
-    modify(Tag.StorageMediaFileSetUID, createUid),
-    clear(Tag.StudyID),
-    insert(Tag.StudyInstanceUID, _ => createUid(null)),
-    modify(Tag.SynchronizationFrameOfReferenceUID, createUid),
-    modify(Tag.TargetUID, createUid),
-    modify(Tag.TemplateExtensionCreatorUID, createUid),
-    modify(Tag.TemplateExtensionOrganizationUID, createUid),
-    modify(Tag.TransactionUID, createUid),
-    modify(Tag.UID, createUid),
-    clear(Tag.VerifyingObserverName)))
+  def anonFlow: Flow[DicomPart, DicomPart, NotUsed] = {
+    val sopInstanceUID = createUid(null)
```

Comment:
Made sure same sop instance uid used for both SOPInstanceUID and MediaStorageSOPInstanceUID